File: backend/src/main/java/com/gfi/parkplatzapp/backend/persistence/entities/Parkhaus.java
package com.gfi.parkplatzapp.backend.persistence.entities;

import com.gfi.parkplatzapp.backend.utils.AktivitaetEnum;
import jakarta.persistence.*;
import lombok.*;

import java.util.List;

@Entity
@AllArgsConstructor
@NoArgsConstructor
@ToString
@Getter
@Setter
public class Parkhaus {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long parkhausID;

    private String bezeichnung;
    private String strasse;
    private int hausnummer;
    private int plz;
    private String ort;

    @Enumerated(EnumType.STRING)
    private AktivitaetEnum aktivitaet;

    @OneToMany
    private List<Parkflaeche> parkflaecheList;
}

File: frontend/src/app/facade/Parkhaus.ts
import { AktivitaetsEnum } from "../utils/aktivitaetEnum.utils"
import { Parkflaeche } from "./Parkflaeche"

export interface Parkhaus {
    parkhausID: number
    bezeichnung: string
    straße: string
    hausnummer: number
    plz: number
    ort: string
    aktivitaet: AktivitaetsEnum;
    parkflaecheList: Parkflaeche[]
}